languages/csharp/concept-exercises/pattern-matching/.docs/instructions.md

```diff
@@ -0,0 +1,52 @@
+﻿# Instructions
```

Comment:
Currently, the instructions spell out what the student is expected to do, even with some very concrete examples. However, we want our exercises to designed such that students will be guided to using the correct approach, but not by explicitly stating what they must do. In other words, when solving the exercise, we should design it that the students are highly likely to solve the issue in the way that teaches what Concepts we are trying to teach.

This can be quite tricky to get right, and is probably the most difficulty part of designing an exercise. In general, I'd like this exercise to focus just on the modern, C# 8 pattern matching as described [here](https://docs.microsoft.com/en-us/dotnet/csharp/pattern-matching). This means teaching:

1. The `is` type pattern expression.
1. The pattern matching `switch` statement.
1. The `when` clause in a `switch` statement.
1. The `var` expression in a `case` expression.
1. `switch` expressions

I think this is really the type of discussion we should be having in the GH issue, so I'll also add my comments there.